### PR TITLE
PR Processing data

### PR DESCRIPTION
**Pull request by DarioAlessioR**
PR for branch processing_data, for activity Blog App - processing data in models.

This branch includes the code that sets up associations between models.

It also contains 5 methods created for the 4 classes in the app, according to the following requirement:

-Users

A method that returns the 3 most recent posts for a given user.

-Posts

A method that updates the posts counter for a user.

A method which returns the 5 most recent comments for a given post.

-Comments

A method that updates the comments counter for a post.

-Likes

A method that updates the likes counter for a p

Thank you in advance for your review.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
```
Comment (mateo951) [inline app/models/user.rb:1]: Has User, Post, Comment and Like models inside app/models/ directory 🙌

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
```
Comment (mateo951) [inline app/models/post.rb:4]: All models have the associations set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) 🚀 Perfect!

File: app/models/post.rb [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+
+  after_save :update_post_counter
+
+  def recent_five_comments
+    Comment.last(5)
+  end
+
+  def update_post_counter
+    author.increment!(:post_counter)
+  end
```
Comment (mateo951) [inline app/models/post.rb:14]: Post model has a method that updates the posts counter for a user 🙌

File: app/models/post.rb [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  def recent_five_comments
+    Comment.last(5)
+  end
```
Comment (mateo951) [inline app/models/post.rb:10]: Post model has a method that returns the 5 most recent comments for a given post 👌

File: app/models/like.rb [hunk 5]
```diff
@@ -1,4 +1,10 @@
 class Like < ApplicationRecord
   belongs_to :post
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+
+  after_save :update_likes_counter
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (mateo951) [inline app/models/like.rb:9]: Like model has a method that updates the likes counter for a post 😉 Nicely done!

**Review by mateo951 — APPROVED** [review]
### Hi @DarioAlessioR, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 5 above)
Comment (DarioAlessioR) [reply app/models/like.rb:9]: Great review @mateo951 ...thank you.

**Pull request merged**